### PR TITLE
[harvest] do not update empty pivot data

### PR DESCRIPTION
**Pull request by rastislav-chynoransky**
Fixes https://jira.sng.sk/browse/WEBUMENIA-1353

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have updated the [CHANGELOG](../CHANGELOG.md)
- [x] I have requested at least 1 reviewer for this PR
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: app/Harvest/Harvesters/AbstractHarvester.php [hunk 1]
```diff
@@ -20,6 +20,42 @@ public function __construct(AbstractRepository $repository, AbstractImporter $im
...
+                $harvest->status_messages = trans('harvest.status_messages.progress', [
+                    'current' => ++$i,
+                    'total' => $total,
+                ]);
+                $harvest->save();
+            }
+
+            $harvest->status = SpiceHarvesterHarvest::STATUS_COMPLETED;
+            $harvest->completed = date('Y-m-d H:i:s');
+            $harvest->status_messages = trans('harvest.status_messages.finished', [
```
Comment (eronisko) [inline app/Harvest/Harvesters/AbstractHarvester.php:45]: From here, I can't tell the difference between  `harvest.status_messages.finished` and `harvest.status_messages.completed`. Is there any other name (names) we could use?

**Review by eronisko — CHANGES_REQUESTED** [review]
Disclaimer: I haven't wrapped my head around the Harvester > Harvest/Importer > Record architecture yet. (Unfortuntely I'm quite confused by it)

So here are some general thoughts, happy to discuss them further

File: app/Harvest/Harvesters/AbstractHarvester.php [hunk 2]
```diff
@@ -20,6 +20,42 @@ public function __construct(AbstractRepository $repository, AbstractImporter $im
         $this->importer = $importer;
     }
 
+    public function tryHarvestFailed(SpiceHarvesterHarvest $harvest) {
```
[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply app/Harvest/Harvesters/AbstractHarvester.php:23]: Great suggestion! There is obvious repetition of code that we need to refactor. However I'm not sure if the model should handle reporting logic and thus got a dependency to sentry.

[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply app/Harvest/Harvesters/AbstractHarvester.php:23]: But I'm thinking maybe throwing the exception to sentry doesn't belong here at all...

[the remark this answers is not in the document]
Comment (eronisko) [reply app/Harvest/Harvesters/AbstractHarvester.php:23]: Good point. Two ideas off of the top of my head:

1. Firing an Event when a harvest fails: https://laravel.com/docs/6.x/events#registering-events-and-listeners
1. Adding another callback, e.g.
```php
// app\SpiceHarvesterHarvest.php
public function process(Closure $onProcess, Closure $onError) {
```

**Comment by rastislav-chynoransky** [comment]
@eronisko I'm going to write more tests for harvester and will split this issue to `fix` and `feature` as those two are independent.

**Review by eronisko — APPROVED** [review]

**Pull request merged**